packages/rpc-methods/src/restricted/invokeSnap.ts

```diff
@@ -49,35 +84,64 @@ type InvokeSnapSpecification = ValidPermissionSpecification<{
  * and install it if it's not available yet.
  *
  * @param options - The specification builder options.
- * @param options.allowedCaveats - The optional allowed caveats for the permission.
  * @param options.methodHooks - The RPC method hooks needed by the method implementation.
  * @returns The specification for the `wallet_snap_*` permission.
  */
 const specificationBuilder: PermissionSpecificationBuilder<
   PermissionType.RestrictedMethod,
   InvokeSnapSpecificationBuilderOptions,
   InvokeSnapSpecification
-> = ({
-  allowedCaveats = null,
-  methodHooks,
-}: InvokeSnapSpecificationBuilderOptions) => {
+> = ({ methodHooks }: InvokeSnapSpecificationBuilderOptions) => {
   return {
     permissionType: PermissionType.RestrictedMethod,
-    targetKey,
-    allowedCaveats,
+    targetKey: WALLET_SNAP_PERMISSION_KEY,
+    allowedCaveats: [SnapCaveatType.SnapIds],
     methodImplementation: getInvokeSnapImplementation(methodHooks),
+    validator: ({ caveats }) => {
+      if (caveats?.length !== 1 || caveats[0].type !== SnapCaveatType.SnapIds) {
```

Comment:
Shouldn't all the snap IDs be validated here as well?

Reply:
No, because the specification validator is always called in conjunction with the caveat validator